src/service/viewer-impl.js

```diff
@@ -603,14 +613,32 @@ export class Viewer {
 
   /**
    * Returns a Promise that only ever resolved when the current
-   * AMP document becomes visible.
+   * AMP document first becomes visible.
    * @return {!Promise}
    */
   whenFirstVisible() {
     return this.whenFirstVisiblePromise_;
   }
 
   /**
+   * Returns a Promise that resolve when current doc becomes visible
+   * @return {!Promise}
+   */
+  nextVisiblePromise() {
```

Comment:
`whenNextVisible` to match `whenFirstVisible`?

Reply:
done